src/main/java/com/emc/object/s3/LargeFileUploader.java

```diff
@@ -834,6 +888,43 @@ public LargeFileUploader withAbortMpuOnFailure(boolean abortMpuOnFailure) {
         return this;
     }
 
+    public LargeFileUploader withSourceVersionId(String sourceVersionId) {
+        setSourceVersionId(sourceVersionId);
+        return this;
+    }
+
+    private class CopyPartTask implements Callable<MultipartPartETag> {
+        private final String uploadId;
+        private final int partNumber;
+        private final long offset;
+        private final long length;
+        public CopyPartTask(String uploadId, int partNumber, long offset, long length) {
+            this.uploadId = uploadId;
+            this.partNumber = partNumber;
+            this.offset = offset;
+            this.length = length;
+        }
+
+        @Override
+        public MultipartPartETag call() {
+            if (!active.get()) {
+                // we were paused or aborted, so should not start any more tasks
+                throw new CancellationException();
+            } else {
+                log.debug("uploading {}/{}, uploadId: {}, partNumber {} (offset: {}, length: {})",
+                        bucket, key, uploadId, partNumber, offset, length);
+                CopyPartRequest copyPartRequest = new CopyPartRequest(srcBucket, srcKey, bucket, key, resumeContext.getUploadId(), partNumber)
```

Comment:
`copyPartRequest` does not include the source versionId

Reply:
fixed